tests/rptest/services/cluster.py

```diff
@@ -113,20 +129,7 @@ def wrapped(self: HasRedpanda, *args: Any, **kwargs: Any):
                                self.redpanda.logger, t_initial,
                                disk_stats_initial)
 
-                for redpanda in all_redpandas(self):
-                    redpanda.logger.exception(
-                        f"Test failed, doing failure checks on {redpanda.who_am_i()}..."
-                    )
-
-                    # Disabled to avoid addr2line hangs
-                    # (https://github.com/redpanda-data/redpanda/issues/5004)
-                    # self.redpanda.decode_backtraces()
-
-                    if isinstance(redpanda, RedpandaServiceBase):
-                        redpanda.cloud_storage_diagnostics()
-                    if isinstance(redpanda,
-                                  RedpandaService | RedpandaServiceCloud):
-                        redpanda.raise_on_crash(log_allow_list=log_allow_list)
+                _do_crash_checks(self)
```

Comment:
Pulled this out and call it on both exceptional and non-exceptional cases.

Reply:
just for information: this will cause failures to be re-reported i believe due to the way failure signatures are generated (related https://redpandadata.slack.com/archives/C06MCRY75U7/p1741709246332199?thread_ts=1741199852.252909&cid=C06MCRY75U7)